src/features/teams/page/Teams.tsx

```diff
@@ -1,51 +1,28 @@
-import React, { useEffect, useState } from "react";
-
-const apiFetch = async (
-  endpoint: string,
-  options: RequestInit = {}
-): Promise<any> => {
-  const token = localStorage.getItem("token");
-
-  if (!token) {
-    alert("Unauthorized! Please log in.");
-    throw new Error("Unauthorized");
-  }
-
-  const headers = {
-    "Content-Type": "application/json",
-    Authorization: `Bearer ${token}`,
-    ...options.headers,
-  };
-
-  const response = await fetch(`http://localhost:5001/api${endpoint}`, {
-    ...options,
-    headers,
-  });
-
-  if (!response.ok) {
-    const errorData = await response.json();
-    throw new Error(errorData.msg || "Something went wrong");
-  }
-
-  return response.json();
-};
-
-const Teams: React.FC = () => {
-  const [teams, setTeams] = useState<
-    { id: number; name: string; members: { email: string; username: string }[] }[]
-  >([]);
+import { useEffect, useState } from "react";
+import { addMemberToTeam, createTeam, deleteMemberFromTeam, getTeams } from "../api/teams";
+
+interface Team {
+  id: number;
+  name: string;
+  members: {
+    email: string;
+    username: string;
+  }[];
```

Comment:
```suggestion
  members: User[];
```
Mit dem passenden `User` interface